src/lib/zf/private/stack_alloc.c

```diff
@@ -515,7 +524,16 @@ static int zf_stack_init_nic_capabilities(struct zf_stack* st, int nicno)
     sti->nic[nicno].flags |= ZF_RES_NIC_FLAG_RX_LL;
 
   rc = ef_pd_capabilities_get(dh, pd, dh, EF_VI_CAP_TX_FW_VARIANT, &variant);
-  if( rc != 0 ) {
+  if( rc == -EOPNOTSUPP ) {
+    rc = ef_pd_capabilities_get(dh, pd, dh, EF_VI_CAP_CTPIO_ONLY, &variant2);
+    if( rc != 0 ) {
+      zf_log_stack_err(st, "Unknown TX mode for interface %s (rc %d)\n",
+                           sti->nic[nicno].if_name, rc);
+      return -EOPNOTSUPP;
+    }
+    sti->nic[nicno].flags |= ZF_RES_NIC_FLAG_CTPIO_ONLY;
+  }
+  else if( rc != 0 ) {
```

Comment:
The same here - we shouldn't conflate reporting of FW variant with other capabilities. These should be checked independently.